mcs/class/corlib/Test/System.Reflection/MonoGenericClassTest.cs

```diff
@@ -201,6 +203,70 @@ public void BaseTypeMustReturnNonInflatedType ()
 			Assert.AreSame (expected, t.BaseType, "#1");
 			
 		}
+
+		[Test]
+		public void GenericClassFromStaleTypeBuilderDoesNotClassInit ()
+		{
+			// interface JJJ<T> {
+			//   abstract void W (x : T)
+			// }
+			MethodInfo winfo = null;
+			TypeBuilder ib = null;
+			Type ic = null;
+			Type icreated = null;
+			{
+				ib = module.DefineType ("Foo.JJJ`1",
+							 TypeAttributes.Public
+							 | TypeAttributes.Interface
+							 | TypeAttributes.Abstract);
+				String[] gens = { "T" };
+				GenericTypeParameterBuilder[] gbs = ib.DefineGenericParameters (gens);
+				var gb = gbs[0];
+				winfo = ib.DefineMethod ("W",
+							 MethodAttributes.Public |
+							 MethodAttributes.Abstract |
+							 MethodAttributes.Virtual,
+							 CallingConventions.HasThis,
+							 typeof(void),
+					 new Type[] { gb });
+				icreated = ib.CreateType();
+
+			}
+
+			// class SSS : JJJ<char> {
+			//   void JJJ.W (x : T) { return; }
+		        // }
+			TypeBuilder tb = null;
+			MethodBuilder mb = null;
+			{
+				tb = module.DefineType ("Foo.SSS",
+							TypeAttributes.Public,
+							null,
+							new Type[]{ icreated.MakeGenericType(typeof(char)) });
+
+				mb = tb.DefineMethod ("W_impl",
+						      MethodAttributes.Public | MethodAttributes.Virtual,
+						      CallingConventions.HasThis,
+						      typeof (void),
+						      new Type[] { typeof (char) });
+				{
+					var il = mb.GetILGenerator ();
+					il.Emit (OpCodes.Ret);
+				}
+			}
+
+			ic = ib.MakeGenericType(typeof (char)); // this is a MonoGenericMethod
+			var mintf = TypeBuilder.GetMethod(ic, winfo);
+			// the next line causes mono_class_init() to
+			// be called on JJJ<char> when we try to setup
+			// the vtable for SSS
+			tb.DefineMethodOverride(mb, mintf);
+
+			var result = tb.CreateType();
+
+			object o = Activator.CreateInstance(result);
+			Assert.IsNotNull(o, "#1");
```

Comment:
Thanks.  The crash in the bug was not even a `CreateInstance`, it was `GetUninitializedObject`, so I thought invoking the default constructor would be enough.  I guess since the bug is about vtables it would be worth actually calling a virtual method to make sure everything is set up properly.